src/main/java/com/ericsson/gerrit/plugins/eiffel/listeners/AbstractEventListener.java

```diff
@@ -0,0 +1,131 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractEventListener.class);
+
+    @Inject
+    private com.google.gerrit.server.config.PluginConfigFactory pluginConfigFactory;
+
+    @Inject
+    @CanonicalWebUrl
+    private String gerritUrl;
+
+    protected final String pluginName;
+    protected final File pluginDir;
+
+    public AbstractEventListener(final String pluginName, final File pluginDir) {
+        this.pluginName = pluginName;
+        this.pluginDir = pluginDir;
+    }
+
+    /**
+     * Based on project name from the gerrit event, creates a project specific
+     * EiffelPluginConfiguration.
+     *
+     * @param gerritEvent
+     * @return EiffelPluginConfiguration
+     */
+    public EiffelPluginConfiguration createPluginConfig(final Event gerritEvent) {
+        ChangeEvent changeEvent = (ChangeEvent) gerritEvent;
+        final Project.NameKey projectNameKey = changeEvent.getProjectNameKey();
+        final EiffelPluginConfiguration pluginConfig = new EiffelPluginConfiguration(pluginName,
+                projectNameKey, pluginConfigFactory);
+        pluginConfig.setPluginDir(pluginDir);
+        return pluginConfig;
+    }
+
+    /**
+     * Returns a boolean indicating if Eiffel event sending is enabled for this project and branch.
+     *
+     * @param gerritEvent
+     * @param pluginConfig
+     * @return boolean
+     */
+    public boolean isEiffelEventSendingEnabled(final Event gerritEvent,
+            final EiffelPluginConfiguration pluginConfig) {
+
+        final ChangeEvent changeEvent = (ChangeEvent) gerritEvent;
+        final String project = changeEvent.change.get().project;
+        if (!isPluginEnabled(pluginConfig, project)) {
+            return false;
+        }
+
+        final String branch = changeEvent.change.get().branch;
+        final String filter = pluginConfig.getFilter();
+        boolean eiffelEventsShouldBeSent = isBranchNameInProjectFilterConfig(branch, filter,
+                project);
+        return eiffelEventsShouldBeSent;
+    }
+
+    protected abstract boolean isExpectedGerritEvent(Event gerritEvent);
+
+    protected abstract void prepareAndSendEiffelEvent(Event gerritEvent,
+            EiffelPluginConfiguration pluginConfig);
+
+    private boolean isPluginEnabled(final EiffelPluginConfiguration pluginConfig,
+            final String project) {
+        if (!pluginConfig.isEnabled()) {
+            LOGGER.debug("Eiffel plugin is disabled for project '{}'.\n"
+                    + "Please refer to Eiffel plugin documentation to find out how to configure and enable plugin\n"
+                    + "{}plugins/{}/Documentation/index.html", project, gerritUrl, pluginName);
+            return false;
+        }
+
+        return true;
+    }
+
+    private boolean isBranchNameInProjectFilterConfig(final String branch, final String filter,
+            final String project) {
+        if (filter == null || filter.isEmpty()) {
```

Comment:
This logic sounds a bit wierd if you look at the method name. "isBranchNameInProjectFilterConfig". Here we clearly see that it isn't but we are returning true. How is this filtering suppose to work?

Reply:
You may find the filter functionality in the documentation.

But in short, if filter is added, only pranches in that filter will be enabled. example regex: (dev).* will enable event sending for all branches that start on 'dev'. If filter is null or empty it should send for all branches.

Reply:
Then these lines do still not belong here I think. You should check that outside this method. This method should only check if something is in the filter or not. It's an entire different condition (if nothing was entered, do no filtering)

Reply:
Better?